UI/MapUI.gd

```diff
@@ -13,46 +13,52 @@ func _on_return_button_press():
 	queue_free()
 
 
+var _ROOM_TEXTURE_INDEX = 0
+var _COLOR_RECT_INDEX = 0
+var _SCROLL_CONTAINER_INDEX = 1
+var _ROOM_CONTAINER_INDEX = 0
+
 func _ready():
-	var current_map = MapManager.current_map
+	var current_map: MapBase = MapManager.current_map
 	
 	# Create New Room Object to append to the room container
-	var new_room = room_ui.instantiate()
-	var new_room_size = new_room.get_child(0).get_size()
+	var new_room: Control = room_ui.instantiate()
+	var new_room_size: Vector2 = new_room.get_child(_ROOM_TEXTURE_INDEX).get_size()
 	
 	# Get the objets of the Map UI Scene
-	var color_rect = get_child(0)
-	var scroll_container = get_child(1)
-	var room_container = scroll_container.get_child(0)
+	var color_rect: ColorRect = get_child(_COLOR_RECT_INDEX)
+	var scroll_container: SmoothScrollContainer = get_child(_SCROLL_CONTAINER_INDEX)
+	var vertical_scroll_bar: VScrollBar = scroll_container.get_v_scroll_bar()
+	var room_container: ColorRect = scroll_container.get_child(_ROOM_CONTAINER_INDEX)
```

Comment:
This is fine, but just to provide one more possible alternative to `get_child()` you could do
```GDScript
@export var _scroll_container: SmoothScrollContainer = null
@export var _room_container: ColorRect = null
@export var _color_rect: ColorRect = null
```
at the top of this script. Then the Godot inspector will let you select children of the hierarchy.

Reply:
I didn't get to refactoring out those `get_child()` shenanigans, but I think I'll go down the @export route, just easier to read 😂